pkg/sql/sem/tree/backup.go

```diff
@@ -139,10 +140,22 @@ var _ NodeFormatter = &RestoreOptions{}
 type Restore struct {
 	Targets            TargetList
 	DescriptorCoverage DescriptorCoverage
-	From               []StringOrPlaceholderOptList
-	AsOf               AsOfClause
-	Options            RestoreOptions
-	Subdir             Expr
+
+	// From contains the URIs for the backup(s) we seek to restore.
+	//   - len(From)>1 implies the user explicitly passed incremental backup paths,
+	//     which is only allowed using the old syntax, `RESTORE <targets> FROM <destination>.
+	//     In this case, From[0] contains the URI(s) for the full backup.
+	//   - len(From)==1 implies we'll have to look for incremental backups in planning
+	//   - len(From[0]) > 1 implies the backups are locality aware
+	//   - From[i][0] must be the default locality.
```

Comment:
Is this true elsewhere? i.e. during BACKUP can the =default locality be in any position?

Reply:
it's not true for BACKUP, as the default locality there is resolved using `getURIsByLocalityKV` in backup_planning.go, whereas in RESTORE, `resolveBackupManifests` assumes the first uri is default.... I'll create an issue related to refactoring destination resolution code in B/R planning.